docs: External Audit Storage

nklaassen opened the pull request:
This commit adds documentation for External Audit Storage, a Teleport Enterprise Cloud feature.

nklaassen committed:
docs: External Audit Storage

nklaassen commented on docs/pages/choose-an-edition/teleport-cloud/external-audit-storage.mdx:
```diff
@@ -0,0 +1,130 @@
+
+Currently it is a requirement that the AWS infrastructure is hosted in the same
+AWS region as the Teleport Auth Service for your cluster.
+Historically this has always been us-west-2, but new clusters may be deployed to
+other regions in the future.
```
Not sure if we want to "leak" this here

zmb3 replied:
It's fine. We already mention us-west-2 in a few other places.

nklaassen commented on docs/config.json:
```diff
@@ -63,6 +63,11 @@
             },
+            {
+              "title": "External Audit Storage",
+              "slug": "/choose-an-edition/teleport-cloud/external-audit-storage/",
```
Not sure about this path but idk where else to put it :shrug:

nklaassen committed:
add section get started

nklaassen committed:
mention release version

ptgott committed:
Edits for readability

- Clarify the introduction
- Move all prerequisites into a single section
- Move all how-to instructions into two "Step n/d" sections.
- Move the IAM policy example into a `Details` box.
- Separate the migration section to clarify the difference between
  enabling and disabling the feature.
